packages/app-check/src/internal-api.ts

```diff
@@ -76,10 +76,27 @@ export async function getToken(
    * return the debug token directly
    */
   if (isDebugMode()) {
+    // Check for an in-memory debug token from the exchange endpoint.
+    const debugState = getDebugState();
+    if (debugState.exchangeToken && isValid(debugState.exchangeToken)) {
+      return { token: debugState.exchangeToken.token };
+    }
+    // Check for a indexedDB cached debug token from the exchange endpoint.
+    const cachedDebugToken = await readTokenFromStorage(app, 'debug');
+    if (cachedDebugToken && isValid(cachedDebugToken)) {
+      // Write to debug state.
+      debugState.exchangeToken = cachedDebugToken;
+      return { token: cachedDebugToken.token };
+    }
```

Comment:
The naming is very confusing. We are using "debug token" to mean different things here: 1. the debug token used to exchange for an appcheck token 2. the app check token exchanged from a debug token.
 But "debug token" should only mean 1.

Now thinking about it again after seeing the code,  I don't think we need to cache the appcheck token exchanged from a debug token differently. It's just an appcheck token and the backend doesn't care about how it was created as long as it's valid.

That means we just cache the appcheck token normally in the debug mode the same as in the prod mode.

Reply:
Done.